src/Rules/Classes/UnusedConstructorParametersRule.php

```diff
@@ -45,6 +45,12 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
+		foreach ($node->getClassReflection()->getInterfaces() as $interface) {
+			if ($interface->hasMethod('__construct')) {
```

Comment:
`hasConstructor` would be more precise

Reply:
And above instead of `strtolower($method->getName()) !== '__construct'` we should use `$method->isConstructor()`.

Reply:
@ondrejmirtes rebased and updated